sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,755 @@
+SYCL RT may also, when possible, look at all asynchronous malloc and free
+commands and their dependencies in order to more efficiently schedule memory
+allocations. If the memory pool does not have the requirement memory available
+to allocate for the asynchronous malloc command it will then look to allocate
+from the source to the memory pool to provide the required memory.
+
+When an asynchronous free command is executed it will free up memory in the 
+memory pool to be used for a later asynchonrous malloc command. Memory is not
+freed back to the source at this point.
+
+When a SYCL queue which enqueued asynchronous malloc commands is synchronized
+with, i.e. when `queue::wait` or `queue::wait_and_throw` is called, the SYCL RT
+will check the current total memory allocated to the memory pool against the
+memory pool's deallocation threshold. If the total memory allocated is larger
+than than the threshold then the memory will be freed back to the source,
+otherwise it will not.
+
+[_Note:_ An implementation should maintain memory allocated to a memory pool to
+the size specified by the deallocation threshold when possible, however, an
+implementation is permitted to release this memory back to the source if it is
+required by another memory pool or an application in another process.
+_{endnote}_]
+
+
+== Implementation notes
+
+It is expected that for L0 this extension will be implemented within the L0
+adapter, by reserving allocations for the memory pool and opportunistically
+re-using the memory allocated based on the command lists being enqueued to the
+L0 driver.
+
+It is expected that for CUDA this extension will be implemented by mapping onto
+the CUDA stream-ordered allocator feature.
+
+
+== Issues
+
+. Should we allow mixing asynchronous and synchornous memory commands?
++
+--
+*UNRESOLVED*: CUDA allows memory allocated with the asynchronous malloc command
+to be freed with the regular synchronous free command, should we extend this
+capability to SYCL?
+--
+
+. Should we allow freeing memory with a different queue?
++
+--
+*UNRESOLVED*: Should we allow a memory allocation allocated with an asynchronous
+malloc command from one queue to be freed by an asynchronous free command from
+another queue?
+--
+
+. How should we treat an asynchronous malloc command with a memory pool that is
+not associated with that queue?
++
+--
+*UNRESOLVED*: If an asynchronous malloc command is enqueued with a memory pool
+that is not associated with the queue that the command is enqueued from, should
+this result in an error?
+--
+
+. Should we have a default memory pool for `usm::shared`?
++
+--
+*UNRESOLVED*: Currently the proposed API means that there cannot be default
+memory pool for allocations of allocation kind `usm::shared`, and therefore a
+user must create their own explicit memory pool to do so. Is this reasonable or
+should we extend the API to include a default memory pool for allocations of
+allocation kind `usm::shared`?
```

Comment:
I saw this issue after making my comment above about the memory pool for "shared".  Why wouldn't we do this?  It seems weird to provide default memory pools for "host" and "device" but not "shared".